### ISSUE TITLE
[bench request]: Add Manchego v2.1 (Qwen3.5-4B + LoRA, native option probabilities, TypeSafe wire format)

### ISSUE THREAD

**Issue by nschlaepfer** [fstandhartinger/jevbench#56]
Please add **Manchego v2.1** to JevBench.

## The system

- **Model:** [`oraculumai/Manchego`](https://huggingface.co/oraculumai/Manchego) at revision `77403228b7dfdf823af99a5f562bdcf80b708d4c` (tag `v2.1`). These are merged weights: Qwen3.5-4B plus a LoRA, one forward pass per question, with a native probability over the offered options for choice, noul and score.
- **Base:** [`Qwen/Qwen3.5-4B`](https://huggingface.co/Qwen/Qwen3.5-4B) (Apache-2.0).
- **Licence:** the Manchego weights and serving code are Apache-2.0.

## How to run it

The serving code is public at [`nschlaepfer/manchego-serve`](https://github.com/nschlaepfer/manchego-serve), tag [`v0.1.0`](https://github.com/nschlaepfer/manchego-serve/releases/tag/v0.1.0). It serves TypeSafe's wire contract (`POST /v1/systemone`), so the stock `typesafe` adapter should work with no code PR.

```bash
git clone --branch v0.1.0 https://github.com/nschlaepfer/manchego-serve && cd manchego-serve
docker build -t manchego-serve:2.1-cuda .     # the only step that uses the network: fetches the pinned weights, checks their SHA-256
docker run --rm --gpus all -p 127.0.0.1:8000:8000 manchego-serve:2.1-cuda
curl -s http://127.0.0.1:8000/healthz         # weights_verified, support_files_verified, warmup.repeat_identical
```

- **Settings:** the policy behind the model card's numbers, fixed in the server.
  - Contract `auto`: the short prompt up to 26 options, the state-first prompt for 27–255.
  - One forward pass per question, sequential.
  - Temperature 1.0; the client's option order; no test-time ensembling.
  - `confidence` = (K·max p − 1)/(K − 1).
- **At run time:** no network (the server forces `HF_HUB_OFFLINE=1`) and no telemetry.
- **Limits:**
  - up to 255 options, 32,768 prompt tokens per question, and 1,024 questions per request;
  - anything beyond a limit gets an HTTP 422 whose message says so (e.g. "options per choice", "maximum context length").
- **Hardware:** one GPU. The bf16 weights take about 9.3 GB; long prompts need more. The Qwen3.5 linear-attention layers use Transformers' reference kernels, with no extra CUDA packages.
- **Tested so far:** PyTorch on CPU (Docker, no network) and MLX on Apple silicon, with prompts byte-identical to the research server's. **The CUDA image has not yet run on an NVIDIA GPU.** Our first GPU run is scheduled today, and we'll comment here if anything needs a fix before you run it.
- **Sealed and held-out items:** we don't operate an endpoint. Please run it on your own hardware.

## Disclosures

- **Training labels** are computed by code or are original human annotations (public datasets and Super-NaturalInstructions tasks). No outputs from Jev (TypeSafe AI's hosted service) were used as training targets.
- **JevBench-informed:** our training families were designed from an earlier, unreleased version's per-family hard-tier scores on the **public** items (the published spec only). No item text is in the training data, and every generator is checked to share no 8-word sequence with the public items.
- **Public items during development:** we also read results on the 231 public items for reporting and for comparing versions.
- **Jev elsewhere:**
  - An early audit used Jev to flag phrases in two template data pools (86 phrases dropped).
  - The first training stage's checkpoint selection read one development group whose targets came from Jev. A rule that excludes that group selects the same checkpoint.
- **Sanity reference, not a score:** in our own run of the 231 public items (our harness, one pass), v2.1's accuracy was 0.805. That's only a check that your run is behaving; we know public-only numbers are not a JevBench score.

Model card with every result, weakness and caveat: https://huggingface.co/oraculumai/Manchego

**Comment by fstandhartinger** [comment]
Thanks — this is logged for reachability checks and the JevBench measurement queue. We’ll only make benchmark claims after our own run.

- Harold

**Comment by nschlaepfer** [comment]
Follow-up, as promised: **please use tag `v0.1.1`**. I've updated the issue body.

- **What broke in v0.1.0:** its CUDA Docker build failed. The pinned `pytorch/pytorch` CUDA base is Ubuntu 24.04, whose system Python is marked externally managed (PEP 668), so `pip install` refused.
- **The fix:** v0.1.1 adds one line (`ENV PIP_BREAK_SYSTEM_PACKAGES=1`). The served model, prompts, settings and outputs are unchanged.

**First NVIDIA run of v0.1.1's recipe** (Lambda A10 24 GB, driver 570.148.08, `docker build` then `docker run --gpus all`):
- **Build:** 76 s, including the pinned weight download and SHA-256 check.
- **`/healthz`:** healthy; `weights_verified: true`, `support_files_verified: true`, `warmup.repeat_identical: true`.
- **Fixture requests:** all 40 answered, with 0 argmax changes against our MLX 8-bit reference. The largest probability difference was 0.070, on the 255-option requests; at most 0.029 elsewhere.
- **Long input:** a 7,511-token request answered in 2.0 s.
- **Serial latency** (one request at a time, loopback): p50 81 ms, p95 82 ms per single-question request, with Transformers' reference linear-attention kernels.
- **Peak GPU memory:** 13.5 GB, so any GPU with 16 GB or more works.

**For your run:**
- **Cost basis:** the hosted Qwen3.5-4B list price (DeepInfra, $0.03 per million input tokens). The model produces 0 output tokens, and the server reports `usage.input_tokens` for every request.
- **Timing:** we'd suggest loopback on the same machine, one request at a time, as above.
- **Without Docker:** `pip install torch==2.10.0 --index-url https://download.pytorch.org/whl/cu128 && pip install . && manchego-serve-download && manchego-serve --backend torch` (README, Quick start).

As before, our 0.805 on the public items is only a sanity reference, not a score.

**Comment by fstandhartinger** [comment]
Received via X DM to @airesearch12 on 23 Sep 2026: https://x.com/i/chat/1579783530107883521-1616477899242221568. Logged here for the measurement queue; this adds no benchmark result or claim.
